pkg/ddl/reorg.go

```diff
@@ -289,11 +289,6 @@ func (w *worker) runReorgJob(reorgInfo *reorgInfo, tblInfo *model.TableInfo,
 		if err != nil {
 			return errors.Trace(err)
 		}
-	case <-w.ctx.Done():
```

Comment:
I see that we rely on doneCh to find w.ctx.Done(). How about change the signature of `f` to have a context parameter, and in runReorgJob use `w.ctx` as argument. So we have more confident to remove this `case <-w.ctx.Done():`

Reply:
all current `f`s also calls through `worker`, it should be fine except that we still uses `background` context inside, will change it later